syntax = "proto3";

message BlockMessage {
  string nonce = 1;
  string previous_hash = 2;
  string hash = 3;
  string calculated_hash = 4;
  string model_type = 5;
  string storage_reference = 6;
  int32 block_number = 7;
}

message BlockResponse {
  bool success = 1;
  string message = 2;
}

message BlockValidation {
  bool valid = 1;
}

message ValidationResponse {
  bool success = 1;
}

message ClientMessage {
  bytes value = 1;
  int32 dataset_number = 2;
  string client_id = 3;
}

message NodeResponse {
  bool success = 1;
}

service NodeService {
  rpc AddWeightsFromClient (ClientMessage) returns (NodeResponse);
  rpc AddBlockRequest (BlockMessage) returns (BlockResponse);
  rpc AddBlockToChain (BlockValidation) returns (ValidationResponse);
}